### ISSUE TITLE
I would like a run once at very beginning right before Qunit starts testing

### ISSUE THREAD

**Issue by dtracers** [jquery/qunit#726]
Setup:
My specific case has to deal with the unit testing of HTML5 custom elements.

When I create a HTML5 custom element I typically have some javascript backend that handles some functionality.  For example lets say I have a custom element that have a custom element to take in a number does some fancy math and changes some html internally.

here is what the js side would look like.
```
function CustomElement1() {
}
CustomElement1.prototype = Object.create(HTMLElement.prototype);
CustomElement1.prototype.fancyFunction = function(number) {
 // do fancy stuff here and interaction with some shadow DOM
};
```

I am using HTML imports to load and register the custom elements so it will be set up and ready to use with one line of import code

I am also using a code coverage library (http://blanketjs.org/) that works in tandem with QUnit.

Blanket modifies the javascript file to perform code coverage.
But I think it can only modify code in a QUnit instance.  By that I mean only js code run inside a QUnit code is registered by blanket.

tl;dr using custom elements and code coverage library.

Constraints:
Registering a custom element can only happen once per a page and must happen before any tests make use of the element.
To make blanket work I must register the element inside of a Qunit method

Problem:
Right now I use the Qunit setupOnce inside modules to initialize the element and it works great!
BUT as a result I can only use one module in my test code instead of multiple modules.  (Or add some other setup that makes it call only once regardless of test order)

I noticed that you guys are moving to a 2nd version of QUnit to so it would be really nice if you could either support custom element registration that can occur after reporters do some tooling.
Or if you had some advice on the best way to mitigate this problem with the best coding style.

Thanks!  I love your library!

**Comment by JamesMGreene** [comment]
> To make blanket work I must register the element inside of a Qunit method

Can you explain why that is? I haven't worked with BlanketJS before.

Would you be able to register your custom element in a [`QUnit.begin` callback function](http://api.qunitjs.com/QUnit.begin/)? That seems like the right mechanism for this kind of task.

**Comment by dtracers** [comment]
as far as my (limited) understanding of how the library works it modifies my javascript to insert a line of code in between every line of my code to figure out what lines were run by my tests and which lines were not run by my tests.

And I just tried using QUnit.begin and it did not work.

**Comment by jzaefferer** [comment]
Can you put together a test page that demonstrates the problem?
